isaactest/tests/board_builder.py

```diff
@@ -0,0 +1,163 @@
+        save_button.click()
+        time.sleep(WAIT_DUR)
+        alert = driver.switch_to.alert
+        assert 'save' in alert.text, 'Alert text did not contain "save"'
+        alert.accept()
+
+        log(INFO, "Use a URL to populate board builder fields")
+        time.sleep(WAIT_DUR)
+        driver.get(ISAAC_WEB + '/game_builder?query="physics_skills_14"&subject=physics&level=any&sort=title')
+        subject_field = Select(driver.find_element_by_xpath('//select[@ng-model="questionSearchSubject"]'))
+        level_field = Select(driver.find_element_by_xpath('//select[@ng-model="questionSearchLevel"]'))
+        query_field = driver.find_element_by_xpath('//input[@ng-model="questionSearchText"]')
+        time.sleep(WAIT_DUR)
+        question_results_text = [question_result.text for question_result in driver.find_elements_by_xpath("//ul[@class='no-bullet results-list ng-scope']//li")]
+        assert subject_field.first_selected_option.text == 'Physics', 'Subject field value "{}" does not match expected "{}"'.format(subject_field.first_selected_option.text, 'Physics')
+        assert level_field.first_selected_option.text == 'Any', 'Level field value "{}" does not match expected "{}"'.format(level_field.first_selected_option.text, 'Any')
+        assert query_field.get_attribute('value') == '"physics_skills_14"', 'Query field value "{}" does not match expected "{}"'.format(query_field.get_attribute('value'), '"physics_skills_14"')
+        assert question_results_text == sorted(question_results_text), 'Question results were not sorted as was expected\n{}'.format(question_results_text)
+
+        log(INFO, "Try to save board with a pre-existing ID")
+        title_field = driver.find_element_by_xpath('//input[@ng-model="currentGameBoard.title"]')
+        title_field.send_keys('Test Duplicate ID Board')
+        question_checkboxes = driver.find_elements_by_xpath("//ul[@class='no-bullet results-list ng-scope']//li//input[@type='checkbox' and @ng-model='enabledQuestions[question.id]']")
+        question_checkboxes[0].click()
+        id_field = driver.find_element_by_xpath('//input[@ng-model="currentGameBoard.id"]')
+        id_field.send_keys(random_id)  # Re-use ID used earlier
+        save_button = driver.find_element_by_xpath('//button[@type="submit" and text() = "Save this board"]')
+        save_button.click()
+        alert = driver.switch_to.alert
+        assert 'save' in alert.text, 'Alert text did not contain "save"'
+        alert.accept()
+        time.sleep(1)
+        assert driver.find_elements_by_xpath('//h4[text() = "Save Operation Failed"]')[0].is_displayed(), 'Error not displayed when expected after saving with same ID'
+
+        log(PASS, "Board builder functions as expected.")
+        return True
+
+    except AssertionError as e:
+        log(ERROR, "Asserton Error: {}".format(e))
+        return False
+    except Exception as e:
```

Comment:
Please don't ever do this. Catch specific named exceptions in tests, but don't swallow all and every exception. Unplanned exceptions should propagate up to the main "try...except...finally" block, since unexpected exceptions are usually the site being broken or something more serious than this single test having an issue.

Also, `.format(...)` is misspelt on the line below.